lib/modulesync.rb

```diff
@@ -173,8 +175,13 @@ def self.update(cli_options)
     managed_modules.each do |puppet_module|
       begin
         manage_module(puppet_module, module_files, defaults)
-      rescue # rubocop:disable Lint/RescueWithoutErrorClass
-        warn "Error while updating '#{puppet_module.given_name}'"
+      rescue ModuleSync::Error, Git::GitExecuteError => e
+        message = e.message || "Error during '#{options[:command]}'"
+        $stderr.puts "#{puppet_module.given_name}: #{message}"
+        exit 1 unless options[:skip_broken]
+        errors = true
```

Comment:
Should this instead be:
```suggestion
        errors = true
        break unless options[:skip_broken]
```

That way the error handling at the end of the method is called (and it respects `options[:fail_on_warnings]`).

Reply:
No, IMHO, it should not be like this... The trouble comes from distinguish errors and warnings.

Again, this part could be reworked better if we had more behavior tests.

Reply:
From my point of view, we should change our approach about this:
 * `skip_broken` is not relevant option as, for example, an "update" that does not contain modification is consider as broken
 * `skip_broken` sounds me to be the default so turning option from `skip_broken` to `fail_fast` makes me sense to me
 * `fail_on_warnings` could be an option if we choose to introduction a distinguishment between real error (e.g. unable to process a template) and a warning (e.g. no files are updated after an `update` run)

Reply:
As it introduces a big breaking change, I prefer to finalize current work, then discuss about it if needed, then purpose a patch.

Reply:
I opened a new issue about this point: #218